buf.gen.yaml

```diff
@@ -13,3 +13,12 @@ plugins:
   - plugin: buf.build/grpc/go:v1.3.0
     out: internal/gen/proto/go
     opt: paths=source_relative
+  - plugin: buf.build/protocolbuffers/js:v3.21.2
+    out: grpcwebclient/gen/proto
+    opt: import_style=commonjs,binary
+  - plugin: buf.build/bufbuild/es
+    out: grpcwebclient/gen/proto/es
+    opt: target=ts
```

Comment:
The gRPC-web client expects code generated by the `protocolbuffers/js` plugin, but the code generated by Protobuf-ES is much easier and more intuitive to work with. So, what I'm actually doing is generating both and only using the `protocolbuffers/js` code in the actual client implementation when it needs to invoke calls and get responses. If there's a better way to do this, I'm down to change it.

Reply:
So the client code you have is mixing both protobuf-es and protocolbuffers-js? That seems a little odd... Is it easy to convert between then in the client code I guess? It might be nicer to just use protocolbuffers-js, just so the setup and the code is a little simpler. But if that API and generated code is really that unpleasant, it's probably worth at least adding some comments -- maybe here in this YAML file but maybe also in a README.md for the grpcwebclient folder that describes its purpose and why the looks the way it does.

Reply:
Unfortunately that's correct. In addition to the unpleasantness of the API there is a bit of a showstopper in that there doesn't appear to be any way to convert an object _from_ JSON. There is only a `toObject` function and we need the ability to convert from JSON because the Webdriver API returns us a JSON string. There's probably ways to work around this, but in the end, it just felt easier to use Protobuf-ES everywhere and only convert to/from protocolbuffers/js when interacting with the gRPC-web client API.

I documented this in a comment in both `buf.gen.yaml` and a new README for the `grpcwebclient` directory. Hopefully this is sufficient.

@timostamm if you are aware of anything I'm missing here, let me know.

Reply:
It is correct that github.com/protocolbuffers/protobuf-javascript does not implement the JSON format. 

I think there are several complications going on that make it a bit difficult to follow what's going on:
1. The jasmine test framework is used, but we're not actually using it to test anything.
2. We're generating code for two different protobuf implementations, so it's a bit muddy what's under test.
3. There's an inversion of control with the jasmine test invoking the `connectconformance` command. 
4. There's IPC between the jasmine test, the `pipe` command, and also IPC with the browser.

I would try to remove the inversion of control (is it buying us anything here?). 

To transmit a test case to the browser and results back, you do not need protobuf JSON. Binary data is just an array of bytes, which is just an array of numbers, which is valid JSON.

Reply:
Ideally, we should be passing `conformance-runner.ts` file to the conformance binary and that .ts file would launch Webdriver, the browser, etc. I actually started down this path when first trying to write this but ran into one problem after another with launching the browser and communicating via `stdin`. In the end, I settled on the approach in https://github.com/connectrpc/connect-es/pull/952 for two reasons: 1. it works and 2. I felt like I was spending too much time tinkering around with everything for something that is arguably not entirely required to test Connect implementations.

I can revisit all of this if we'd rather it work a different way. Just not sure how much time we want to invest in that. I guess at the least I can try to remove the usage of two different Protobuf implementations. 

Let me know what you guys think though.

Reply:
Update: was able to remove Protobuf-ES and strictly use protocolbuffers/js. I'm passing the binary data to and from Webdriver now 👍